content/reference/routing.md

```diff
@@ -0,0 +1,210 @@
+
+## Fallback routes
+Preparing for things to go wrong is a large part of programming. At some point,
+someone using an application will land on an unexpected route. It's important
+to not just crash the page, but to show something helpful to explain what just
+happened. This is where fallback routes come in.
+
+```js
+var html = require('choo/html')
+var choo = require('choo')
+
+var app = choo()
+app.route('/', view)               // 1.
+app.route('/404', notFound)        // 2.
+app.route('/*', notFound)          // 3.
+app.mount('body')                  // 4.
+
+function view () {
+  return html`
+    <body>
+      <a href="/uh-oh">
+        Click Click Click
+      </a>
+    </body>
+  `
+}
+
+function not found () {
+  return html`
+    <body>
+      <a href="/">
+        Route not found. Navigate back.
+      </a>
+    </body>
+  `
+}
+```
+
+1. We define our base view on route `/`. This is the first route that's loaded
+   when a person visits our site. It contains a single anchor tag that points
+   to `/uh-oh`, which is a route that doesn't exist.
+2. It's good practice to define a fallback route statically as `/404`. This
+   helps with debugging, and is often treated specially when deploying to
+   production.
+3. We define our fallback route as `*`. The asterix symbol is pronounced
+   "glob". Our glob route will now handle all routes that didn't match
+   anything.
+4. We mount the application on the DOM. If someone now clicks the link that's
+   rendered in `/`, it will be handled by the fallback route.
+
+## Dynamic routing
+Sometimes there will be pages that have the same layout, but different data.
+For example user pages, or blog entries. This requires _dynamic routing_. In
+Choo we have two types of syntax for dynamic routing.
+
+### Params
+Params are declared with the `:` syntax. For example `/foo/:bar`. This means
+that the `/foo` part is static, and `:bar` can be any value, up until the next
+slash (`/`).
+
+The value from a param, is exposed in Choo as `state.params`. So say we have
+the route `/foo/:bar`, and we navigate to `/foo/beep`, the value of
+`state.params.bar` will be `'beep'`.
+
+```js
+var html = require('choo/html')
+var choo = require('choo')
+
+var app = choo()
+app.route('/', placeholder)
+app.route('/:user', placeholder)
+app.route('/:user/:repo', placeholder)
+app.route('/:user/settings', placeholder)
+app.route('/404', placeholder)
+
+function placeholder (state) {
+  console.log(state.params)
+  return html`<body>placeholder</body>`
+}
+```
+
+### Wildcards
+In most cases params are the answer for dynamic routing. They're named, readily
+available, and easy to traverse. However, they can't cover cases where the
+amount of slashes in route will be unknown.
+
+Take for example GitHub's code view. To navigate to Choo's `html/raw` API, the
+route is: https://github.com/choojs/choo/blob/master/html/raw.js. In this case,
+everything after `master/` is unknown. The depth and names of the route
```

Comment:
the second sentence here is very confusing, it'd help if you explained what "project" means in this case